### PR TITLE
Report for flakes on slack

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

As a medic when `main` build fails we often have to check the GHA runs and scroll a long way to find whether a flaky test has failed, etc.

This PR should improve our life a bit to directly see in slack whether a flaky test has occurred and failed the main build. 

Ideally this should allow us also to better track which tests fail more often, as this is currently not really possible.

This means you can see this as a first step to bring back our flaky test statistics.

## Example runs

We distinguish between unit tests and integration tests (in the first iteration) we can always add more or collapse them.

### Flaky parameterized test

When having a parameterized tests, there might be multiple methods and parameters that failed the tests, the new output covers this. Unfortunately we hit here some boundaries of slack messages and line lengths, but I think it is good enough.

run1

### No flakes
If there are no flakes for IT, for example but still for unit tests we can see this as well, and the message doesn't break.

run2

### Only single Unit tests

In most cases it will look like this, where only one unit test has been flaky. The messages is I think good enough to be consumed by a human.

run3

### Test summary

The summary also shows now always the failed flaky test directly.

test-results

### CODE REVIEW

**Comment by Zelldon** [comment]
Break through
break-through

File: .github/actions/analyze-test-runs/action.yml [hunk 1]
```diff
@@ -22,6 +27,43 @@ runs:
...
+          echo 'FLAKY_TESTS<<EOF'
+            # echo ${flakyTests[*]}
+            echo $flakyTests
+          echo EOF
+        } >> $GITHUB_OUTPUT
+
+        echo "::error::Found flaky tests!\n ${flakyTests}"
+      fi
+
   - name: Duplicate test runs
```
Comment (Zelldon) [inline .github/actions/analyze-test-runs/action.yml:67]: 💭 Maybe we don't need the duplicate test run analysis anymore?

File: .github/actions/analyze-test-runs/action.yml [hunk 2]
```diff
@@ -22,6 +27,43 @@ runs:
...
+    env:
+      BUILD_OUTPUT_FILE_PATH: ${{ inputs.buildOutputFilePath }}
+    run: |
+      if [ ! -s "$BUILD_OUTPUT_FILE_PATH" ]; then
+        echo "::error::Build output file does not exist or is empty!"
+        exit 1
+      fi
+
+      # Extracting flaky tests
+      # Based on old Jenkins script
```
Comment (megglos) [inline .github/actions/analyze-test-runs/action.yml:42]: good ol' Jenkins 🫡

**Review by megglos — APPROVED** [review]
❤️  This is such a great improvement!

💭 While you are at it, for then ext iteration, could you extend it to comment on existing flaky test issues or create a new issue? 😄

File: .github/actions/analyze-test-runs/action.yml (hunk 1 above)
Comment (megglos) [reply .github/actions/analyze-test-runs/action.yml:67]: good point not sure if it's really worth it still, I understood it was added back then when there were some redundant test runs by accident, however in worst case we would notice high duration times if this is a big problem and would likely spot that easily anyway?

**Comment by Zelldon** [comment]
>  While you are at it, for then ext iteration, could you extend it to comment on existing flaky test issues or create a new issue?

Ha. Actually, I thought about it, but I felt it was not that easy/quickly doable. At least finding existing ones. But generating new, might be more easier but with a different format then. For example the title would be just the name of the test, the description maybe just the link to the failed build or something.

File: .github/actions/analyze-test-runs/action.yml (hunk 1 above)
Comment (Zelldon) [reply .github/actions/analyze-test-runs/action.yml:67]: @oleschoenburg I think you added them right? Do you have an opinion on this?

**Pull request merged**